Modules/_tkinter.c

```diff
@@ -360,21 +360,36 @@ Sleep(int milli)
 }
 #endif /* MS_WINDOWS */
 
-/* Wait up to 1s for the mainloop to come up. */
+/* Wait for the mainloop to come up. */
+static int mainloopwaitattempts = 10; // 1 second
 
 static int
 WaitForMainloop(TkappObject* self)
 {
     int i;
-    for (i = 0; i < 10; i++) {
-        if (self->dispatching)
-            return 1;
+    if (self->dispatching)
+    {
+        return 1;
+    }
+    for (i = 0; i < mainloopwaitattempts; i++) {
         Py_BEGIN_ALLOW_THREADS
         Sleep(100);
         Py_END_ALLOW_THREADS
+        if (self->dispatching) 
+        {
+            if (PyErr_WarnEx(PyExc_DeprecationWarning,
+                "It seems you are implicitly waiting for "
+                "the mainloop to come up.\n"
+                "This behavior is deprecated; consider polling "
+                "dispatching() instead.\n", 1))
```

Comment:
That is a happy, non-deprecated path when using `_tkinter.tkapp.setmainloopwaitattempts(0)`. This function should warn on timeout too though. I've added a fix by duplicating code, but if you have a better idea, I'm open to it.

Reply:
True, I forgot about `mainloopwaitattempts` being equal to 0. To avoid duplicate code, I would put it where I originally suggested, but just add a check to ensure it is only raised when `mainloopwaitattempts` is not 0.